tests/TestCase/Log/Engine/BaseLogTest.php

```diff
@@ -52,8 +56,36 @@ private function assertUnescapedUnicode(array $needles, $haystack)
      */
     public function testLogUnicodeString()
     {
-        $logged = $this->logger->log(LogLevel::INFO, implode($this->testData));
+        $this->logger->log(LogLevel::INFO, implode($this->testData));
+
+        $this->assertUnescapedUnicode($this->testData, $this->logger->getMessage());
+    }
+
+    public function testPlaceHoldersInMessage()
+    {
+        $context = [
+            'no-placholder' => 'no-placholder',
+            'string' => 'a-string',
+            'bool' => true,
+            'json' => new Entity(['foo' => 'bar']),
+            'array' => ['arr'],
+            'obj' => function () {
+            },
+        ];
+        $this->logger->log(
+            LogLevel::INFO,
+            '1: {string}, 2: {bool}, 3: {json}, 4: {not a placeholder}, 5: {array}, 6: {obj}, 8: {valid-ph-not-in-context}',
```

Comment:
Should we support `.` separated values to create paths? We could use `Hash::get()` to pull values out. I'm also ok with only supporting one dimensional arrays to stay simple.

Reply:
I think simple Should be fine.

Reply:
I too don't think we need to get all fancy here and add dependency on utility package for being able to use `Hash::get()`.